### ISSUE TITLE
Input OnkeyUp with DebounceMilliseconds still produces one event for each keystroke immediately

### ISSUE THREAD

**Issue by michaelcsikos** [ant-design-blazor/ant-design-blazor#2959]
When the `Input` or `Search` has a `DebounceMilliseconds` set, e.g. 475 ms, it is still firing `OnkeyUp` immediately for every keystroke.

The documentation says the `DebounceMilliseconds` should `Delay the processing of the KeyUp event until the user has stopped typing for a predetermined amount of time`.

If the user types `hello` with each keystroke being within 475 ms of the previous keystroke, then after the final keystroke `o` it should then wait 475 ms, and then it should raise the `OnkeyUp` event a single time. Each keystroke should restart the timer.

**Comment by ElderJames** [comment]
Hi @michaelcsikos , do you want `OnkeyUp` to trigger delay as well? The current design is that DebounceMilliseconds only affect the binding of the value, ad `OnKeyUp` still fires events as usual.

**Comment by michaelcsikos** [comment]
@ElderJames, yes, I think `OnkeyUp` should be delayed as well, and that is how I interpret the intent of the documentation.

In my component I have solved this by setting `DebounceMilliseconds = "0"` and using a `System.Threading.Timer`:

```cs
private void OnKeyUp(KeyboardEventArgs e)
{
    if (e.Key == "Escape")
    {
        StopTimer();

        _searchText = null;

        RemoveFilter();
    }
    else
        StartTimer();
}

Timer _timer = null;

private void StopTimer()
{
    _timer?.Dispose();
}

private void StartTimer()
{
    StopTimer();

    _timer = new Timer(new TimerCallback(OnDelayedApplyFilter), null, 475, -1);
}

private void OnDelayedApplyFilter(object timerState)
{
    InvokeAsync(ApplyFilter);
}

public void Dispose()
{
    StopTimer();
}
```

This will correctly call my `ApplyFilter()` method one time only when 475 ms has elapsed after a keystroke. Each keystroke restarts the countdown.

**Comment by buvinghausen** [comment]
I think the documentation should be updated to clarify it debounces the binding

It works great once I wrapped it in a property

```csharp
private string SearchText
{
	get => _searchText;
	set
	{
		if (value == _searchText) return;
		_searchText = value;
		PageNumber = 1;
		OnChangeAsync().GetAwaiter().GetResult();
	}
}
```
